File: backend/app/schemas/audit.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    actor_id: uuid.UUID | None
    actor_name: str | None = None
    action: str
    entity_type: str
    entity_id: uuid.UUID | None
    before: dict[str, Any] | None
    after: dict[str, Any] | None
    at: datetime

File: frontend/src/shared/api/audit.ts
import { api } from "./client";

export interface AuditLogOut {
  id: string;
  actor_id: string | null;
  actor_name: string | null;
  action: string;
  entity_type: string;
  entity_id: string | null;
  before: Record<string, unknown> | null;
  after: Record<string, unknown> | null;
  at: string;
}

export function getAuditLog(params: { action_prefix?: string; entity_type?: string; limit?: number } = {}) {
  const qs = new URLSearchParams();
  if (params.action_prefix) qs.set("action_prefix", params.action_prefix);
  if (params.entity_type) qs.set("entity_type", params.entity_type);
  if (params.limit) qs.set("limit", String(params.limit));
  const suffix = qs.toString() ? `?${qs.toString()}` : "";
  return api.get<AuditLogOut[]>(`/settings/audit-log${suffix}`);
}
